libs/feature/map/src/lib/add-layer-from-file/add-layer-from-file.component.spec.ts

```diff
@@ -0,0 +1,128 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing'
+import { AddLayerFromFileComponent } from './add-layer-from-file.component'
+import { MapFacade } from '../+state/map.facade'
+import { TranslateModule } from '@ngx-translate/core'
+
+class MapFacadeMock {
+  addLayer = jest.fn()
+}
+
+describe('AddLayerFromFileComponent', () => {
+  let component: AddLayerFromFileComponent
+  let fixture: ComponentFixture<AddLayerFromFileComponent>
+  let mapFacade: MapFacade
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [TranslateModule.forRoot()],
+      declarations: [AddLayerFromFileComponent],
+      providers: [
+        {
+          provide: MapFacade,
+          useClass: MapFacadeMock,
+        },
+      ],
+    }).compileComponents()
+
+    mapFacade = TestBed.inject(MapFacade)
+    fixture = TestBed.createComponent(AddLayerFromFileComponent)
+    component = fixture.componentInstance
+    fixture.detectChanges()
+  })
+
+  it('should create', () => {
+    expect(component).toBeTruthy()
+    expect(component.errorMessage).toBeFalsy()
+    expect(component.loading).toBe(false)
+    expect(component.successMessage).toBeFalsy()
+  })
+
+  describe('handleFileChange', () => {
+    describe('should set error message if file is not selected', () => {
```

Comment:
Just a remark in terms of UT naming : 

Usually, we try to reserve the expectations, such as `should do...` to `it` blocks. 

For `describe`  blocks, we try to have either a condition (such as `if file is not selected`), or the name of the section we're testing (eg : `#errorMessage`,like you did line 40 for instance).

So for instance in this case, IMO, the name of the describe block should be `if file is not selected` (and the naming of the following `it` case is alright since it starts with `should`). 

This remark is true for the next few tests in this file :slightly_smiling_face:

Reply:
@cmoinier Thanks for the note on UT naming. I have revised the `describe` blocks for clarity and consistency as suggested.